cmd/chain-maind/app/app.go

```diff
@@ -137,7 +137,24 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		if err != nil {
 			return err
 		}
+
+		tmpFilePath := cleanedPath + ".tmp"
+		tmpFile, err := os.Create(tmpFilePath)
+		if err != nil {
+			return err
+		}
+
 		defer func() {
+			if _, e := os.Stat(tmpFilePath); e == nil {
```

Comment:
Why check stat before remove?

Reply:
Use WriteFile (github.com/google/renameio) to instead